bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/helper/image/AssetDeliveryHelper.java

```diff
@@ -74,8 +71,33 @@ public static String getSrcSet(@NotNull AssetDelivery assetDelivery, @NotNull Re
         return null;
     }
 
-    public static  String getSrc(@NotNull AssetDelivery assetDelivery, @NotNull Resource imageComponentResource, @NotNull String imageName,
-                                 @NotNull String extension, @NotNull int[] smartSizes, int jpegQuality) {
+    public static String getSrcUriTemplate(@NotNull AssetDelivery assetDelivery, @NotNull Resource imageComponentResource,
+                                           @NotNull String imageName, @NotNull String extension,
+                                           @NotNull int[] smartSizes, int jpegQuality, String widthPlaceholder) {
+
+        String templateUrl = getSrc(assetDelivery, imageComponentResource, imageName, extension,
+            smartSizes, jpegQuality, false);
+        if(!StringUtils.isEmpty(templateUrl)) {
+            if(templateUrl.indexOf("?") < 0) {
+                templateUrl += "?" + widthPlaceholder;
+            } else {
+                templateUrl += "&" + widthPlaceholder;
+            }
+        }
+        return templateUrl;
+    }
+
+    public static String getSrc(@NotNull AssetDelivery assetDelivery, @NotNull Resource imageComponentResource,
+                                @NotNull String imageName, @NotNull String extension,
+                                @NotNull int[] smartSizes, int jpegQuality) {
+
+        return getSrc(assetDelivery, imageComponentResource, imageName, extension,
+            smartSizes, jpegQuality, smartSizes.length == 1);
```

Comment:
Shouldn't this be `smartSizes.length > 0`?

Reply:
It adds width only if `smartSizes.length == 1` for non asset delivery. see [here](https://github.com/adobe/aem-core-wcm-components/blob/main/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ImageImpl.java#L272).